src/main/java/com/gluonhq/substrate/util/FileOps.java

```diff
@@ -365,4 +373,127 @@ public static void extractFilesFromJar(String extension, Path sourceJar, Path ta
         }
     }
 
+    /**
+     * Downloads a file from a given URL (non null) into a given path (non null)
+     * @param fileUrl the URL of the file
+     * @param filePath the absolute path of the file where the remote file be downloaded into
+     * @throws IOException
+     */
+    public static void downloadFile(URL fileUrl, Path filePath) throws IOException {
+        Objects.requireNonNull(fileUrl);
+        Objects.requireNonNull(filePath);
+        try (ReadableByteChannel readableByteChannel = Channels.newChannel(fileUrl.openStream());
+             FileOutputStream fileOutputStream = new FileOutputStream(filePath.toFile());
+             FileChannel fileChannel = fileOutputStream.getChannel()) {
+            fileChannel.transferFrom(readableByteChannel, 0, Long.MAX_VALUE);
+        } catch (IOException e) {
+            throw new IOException("Error downloading from " + fileUrl + "into " + filePath + ": " + e.getMessage() + ", " + Arrays.toString(e.getSuppressed()));
+        }
+    }
+
+    /**
+     * Extracts the files from a given zip file into a target folder, and returns a map
+     * with the names of the files and their checksum values.
+     * @param sourceZip the path of a non null zip file
+     * @param targetDir the path of a folder where the zip file will be extracted
+     * @return a map with the file names and their checksum values
+     * @throws IOException
+     */
+    public static Map<String, String> unzipFile(Path sourceZip, Path targetDir) throws IOException {
+        Objects.requireNonNull(sourceZip);
+        Objects.requireNonNull(targetDir);
+        if (!Files.exists(sourceZip)) {
+            throw new IOException("Error: " + sourceZip + " does not exist");
+        }
+        if (!sourceZip.toString().endsWith(".zip")) {
+            throw new IOException("Error: " + sourceZip + " is not a valid zip file");
+        }
+        if (Files.isRegularFile(targetDir)) {
+            throw new IOException("Error: " + targetDir + " is not a directory");
+        }
+        if (!Files.exists(targetDir)) {
+            Files.createDirectories(targetDir);
+        }
+        Map<String, String> hashes = new HashMap<>();
+        try (ZipInputStream zis = new ZipInputStream(new FileInputStream(sourceZip.toFile()))) {
+            ZipEntry zipEntry = zis.getNextEntry();
+            while (zipEntry != null) {
+                File destFile = new File(targetDir.toFile(), zipEntry.getName());
```

Comment:
Probably easier to keep `destFile` as a `Path`

Reply:
ok